GVRf/Framework/framework/src/main/jni/engine/renderer/renderer.cpp

```diff
@@ -334,7 +169,12 @@ void Renderer::cull(Scene *scene, Camera *camera,
     // Note: this needs to be scaled to sort on N states
     state_sort();
 
-    BatchSetup();
+    if(do_batching){
+        if(batch_manager == nullptr){
```

Comment:
In a future change you should consider eliminating this lazy initialization. The batch_manager is needed always by default.